app/src/main/java/com/practicum/playlistmaker/domain/impl/SettingsInteractorImpl.kt

```diff
@@ -0,0 +1,39 @@
+package com.practicum.playlistmaker.domain.impl
+
+import android.content.Context
+import com.practicum.playlistmaker.data.preferences.SettingsManager
+import com.practicum.playlistmaker.domain.api.SettingsInteractor
+import com.practicum.playlistmaker.domain.api.SharedPreferenceRepository
+
+class SettingsInteractorImpl(
+    var repository: SharedPreferenceRepository
```

Comment:
```suggestion
    private val repository: SharedPreferenceRepository
```
Лучше именно через private val передавать в конструктор параметры, т.к. иначе эти классы станут доступны снаружи интерактора, что обесценивает разделение кода на слои

Reply:
> Лучше именно через private val передавать в конструктор параметры, т.к. иначе эти классы станут доступны снаружи интерактора, что обесценивает разделение кода на слои


Спасибо! Исправила.